html/captive-portal/lib/captiveportal/PacketFence/Form/Authentication.pm

```diff
@@ -107,6 +107,35 @@ sub check_aup {
     return ;
 }
 
+=head2 check_telephone_form
+
+Check telephone form
+
+=cut
+
+sub check_telephone_form {
+    my ($self, $field) = @_;
+    if($self->app->request->method eq "POST"){
+        if ( $field->value !~ /^([0-9]+[0-9-\.]+[0-9]+)$/ ) {
+            $field->add_error("Enter a valid telephone <em>number</em>");
+            $self->app->flash->{error} = "Telephone number is not valid";
```

Comment:
I18N?

Reply:
I think its automatically done when rendered in the templates

Reply:
Added in https://github.com/inverse-inc/packetfence/pull/2798/commits/3e1cc8623d9f9e34c743f1fdd36486a80f98034c